keras/src/models/model.py

```diff
@@ -433,6 +433,20 @@ def quantize(self, mode, **kwargs):
         """
         from keras.src.dtype_policies import QUANTIZATION_MODES
 
+        if mode == "gptq":
+            from keras.src.quantizers.gptqconfig import GPTQConfig
+
+            config = kwargs.get("quant_config")
+            if not isinstance(config, GPTQConfig):
+                raise TypeError(
+                    "When using 'gptq' mode, you must pass a `quant_config` "
+                    "keyword argument of type `keras.quantizers.GPTQConfig`."
+                )
+
+            # The config object's own quantize method drives the process.
+            config.quantize(self)
```

Comment:
Based on the other comment in gptqconfig.py, we can remove the quantize method from the config, and rewrite this as follows:

```suggestion
            quantize_model(self, config)
```